src/Compilers/CSharp/Test/Semantic/Semantics/PrimaryConstructorTests.cs

```diff
@@ -16717,5 +16717,862 @@ partial class Point(int i)
+            comp.VerifyEmitDiagnostics();
+        }
+
+        [Fact]
+        public void ShadowedByMemberFromBase_05_WithIdentityConvertedArgument()
+        {
+            var source = @"
+class Base(int p1)
+{
+    protected string p1 = p1.ToString();
+}
+
+class C1(int p1) : Base((int)p1)
+{
+    void M()
+    {
+        string a = p1; 
+    }
+}
+";
+            var comp = CreateCompilation(source, options: TestOptions.ReleaseDll);
+            comp.VerifyDiagnostics();
+            comp.VerifyEmitDiagnostics();
+        }
+
+        [Fact]
+        public void ShadowedByMemberFromBase_06_WithNonIdentityConvertedArgument()
+        {
+            var source = @"
+class Base(long p1)
+{
+    protected string p1 = p1.ToString();
+}
+
+class C1(int p1) : Base(p1)
+{
+    void M()
+    {
+        string a = p1; 
+    }
+}
+";
+            var comp = CreateCompilation(source, options: TestOptions.ReleaseDll);
+
+            var expected = new[] {
+                // (11,20): warning CS9179: Primary constructor parameter 'int p1' is shadowed by a member from base.
+                //         string a = p1; 
+                Diagnostic(ErrorCode.WRN_PrimaryConstructorParameterIsShadowedAndNotPassedToBase, "p1").WithArguments("int p1").WithLocation(11, 20)
+                };
+
+            comp.VerifyDiagnostics(expected);
+            comp.VerifyEmitDiagnostics(expected);
+        }
+
+        [Fact]
+        public void ShadowedByMemberFromBase_07_NotFirstArgument()
+        {
+            var source = @"
+class Base(int p1, int p2)
+{
+    protected string p1 = (p1+p2).ToString();
+}
+
+class C1(int p1) : Base(0, p1)
+{
+    void M()
+    {
+        string a = p1; 
+    }
+}
+";
+            var comp = CreateCompilation(source, options: TestOptions.ReleaseDll);
+            comp.VerifyDiagnostics();
+            comp.VerifyEmitDiagnostics();
+        }
+
+        [Fact]
+        public void ShadowedByMemberFromBase_08_WithParamsArgument()
+        {
+            var source = @"
+class Base(params int[] p1)
+{
+    protected string p1 = p1.ToString();
+}
+
+class C1(int p1) : Base(p1)
+{
+    void M()
+    {
+        string a = p1; 
+    }
+}
+";
+            var comp = CreateCompilation(source, options: TestOptions.ReleaseDll);
+
+            var expected = new[] {
+                // (11,20): warning CS9179: Primary constructor parameter 'int p1' is shadowed by a member from base.
+                //         string a = p1; 
+                Diagnostic(ErrorCode.WRN_PrimaryConstructorParameterIsShadowedAndNotPassedToBase, "p1").WithArguments("int p1").WithLocation(11, 20)
+                };
+
+            comp.VerifyDiagnostics(expected);
+            comp.VerifyEmitDiagnostics(expected);
+        }
+
+        [Fact]
+        public void ShadowedByMemberFromBase_09_ColorColorResolvingToADifferentType()
+        {
+            var source = @"
+class Base
+{
+    protected Type1 Type1 = null;
+}
+
+class C1(int Type1) : Base
```

Comment:
Consider testing when the parameter is passed to `Base`.

Reply:
> Consider testing when the parameter is passed to Base.


I think other tests are adequately covering detection of the fact that a parameter is passed to the base. It feels unnecessary to cover all permutations for unrelated factors.